arangod/Aql/Query.cpp

```diff
@@ -238,7 +239,11 @@ void Query::destroy() {
   exitV8Context();
 
   _snippets.clear();  // simon: must be before plan
-  _plans.clear();     // simon: must be before AST
+
+  TRI_ASSERT(_plans.size() == 1);
```

Comment:
@hkernbach This assertion fails in many tests. Needs investigation. Maybe `_plans.size() <=  1` is more appropriate?

Reply:
@adityamukho yes probably. What has been the actual size? 0? Do you know which test failed and why it could be zero?

Reply:
For testing purpose, we can also get rid of this additional assertion completely.

Reply:
Additionally we then have to handle the case for size: 0 when we decrease memoryusage then as well.